### PR TITLE
Fixed #27378 -- Added UUID serialization support to migration writer

### PR DESCRIPTION
**Pull request by mlorant**
This PR applies the patch made by Yuriy Korobko, availble on the Trac ticket https://code.djangoproject.com/ticket/27378 for 7 days now. It seems complete so I think we can push it forward in the review process.

### CODE REVIEW

**Comment by timgraham** [comment]
I usually try to set the git author when applying someone else's patch. Alternatively, (if we don't know the author's email, for example), "Thanks <name> for the patch." in the commit message is appropriate.

**Comment by mlorant** [comment]
You're right. I append credits to the author in the commit message.

File: django/db/migrations/serializer.py [hunk 1]
```diff
@@ -320,6 +321,11 @@ def serialize(self):
                 return "%s.%s" % (module, self.value.__name__), {"import %s" % module}
 
 
+class UUIDSerializer(BaseSerializer):
+    def serialize(self):
+        return "uuid.UUID('%s')" % self.value, {"import uuid"}
```
Comment (tobiasmcnulty) [inline django/db/migrations/serializer.py]: Could you use `"uuid.%s" % repr(self.value)` here instead, as in the other serializers?

    -        return "uuid.UUID('%s')" % self.value, {"import uuid"}
    +        return "uuid.%s" % repr(self.value), {"import uuid"}

File: tests/migrations/test_writer.py [hunk 2]
```diff
@@ -321,6 +322,29 @@ class IntEnum(enum.IntEnum):
             "default=migrations.test_writer.IntEnum(1))"
         )
 
+    def test_serialize_uuid(self):
+        uuid_a = uuid.UUID('5c859437-d061-4847-b3f7-e6b78852f8c8')
```
Comment (tobiasmcnulty) [inline tests/migrations/test_writer.py]: I would like to see that the round trip works too, e.g.:

    +        self.assertSerializedEqual(uuid.uuid1())
    +        self.assertSerializedEqual(uuid.uuid4())

File: tests/migrations/test_writer.py [hunk 3]
```diff
@@ -321,6 +322,29 @@ class IntEnum(enum.IntEnum):
...
+        uuid_b = uuid.UUID('c7853ec1-2ea3-4359-b02d-b54e8f1bcee2')
+
+        self.assertSerializedResultEqual(
+            uuid_a,
+            ("uuid.UUID('" + str(uuid_a) + "')", {'import uuid'})
```
Comment (tobiasmcnulty) [inline tests/migrations/test_writer.py]: Why not just put the actual string here, as in the model serialization below? E.g.:

	         self.assertSerializedResultEqual(
	             uuid_a,
	-            ("uuid.UUID('" + str(uuid_a) + "')", {'import uuid'})
	+            ("uuid.UUID('5c859437-d061-4847-b3f7-e6b78852f8c8')", {'import uuid'})
	         )
	         self.assertSerializedResultEqual(
	             uuid_b,
	-            ("uuid.UUID('" + str(uuid_b) + "')", {'import uuid'})
	+            ("uuid.UUID('c7853ec1-2ea3-4359-b02d-b54e8f1bcee2')", {'import uuid'})
	         )

This also keeps the test consistent with other tests.

**Comment by mlorant** [comment]
@tobiasmcnulty Thanks for those comments, I applied all of them.

File: django/db/migrations/serializer.py [hunk 4]
```diff
@@ -320,6 +321,11 @@ def serialize(self):
                 return "%s.%s" % (module, self.value.__name__), {"import %s" % module}
 
 
+class UUIDSerializer(BaseSerializer):
+    def serialize(self):
+        return "uuid.%s" % repr(self.value), {"import uuid"}
```
Comment (timgraham) [inline django/db/migrations/serializer.py]: %s -> %r?

Comment (tobiasmcnulty) [inline django/db/migrations/serializer.py]: Other than in error messages, the rest of this file calls `repr()` explicitly, FWIW.

**Pull request merged**